vector/src/main/java/im/vector/app/features/settings/VectorSettingsAdvancedNotificationPreferenceFragment.kt

```diff
@@ -50,29 +51,25 @@ class VectorSettingsAdvancedNotificationPreferenceFragment @Inject constructor()
                         if (newRule != null) {
                             displayLoadingView()
 
-                            session.updatePushRuleActions(
-                                    ruleAndKind.kind,
-                                    preference.ruleAndKind?.pushRule ?: ruleAndKind.pushRule,
-                                    newRule,
-                                    object : MatrixCallback<Unit> {
-                                        override fun onSuccess(data: Unit) {
-                                            if (!isAdded) {
-                                                return
-                                            }
-                                            preference.setPushRule(ruleAndKind.copy(pushRule = newRule))
-                                            hideLoadingView()
-                                        }
-
-                                        override fun onFailure(failure: Throwable) {
-                                            if (!isAdded) {
-                                                return
-                                            }
-                                            hideLoadingView()
-                                            // Restore the previous value
-                                            refreshDisplay()
-                                            activity?.toast(errorFormatter.toHumanReadable(failure))
-                                        }
-                                    })
+                            lifecycleScope.launch {
+                                val result = runCatching {
+                                    session.updatePushRuleActions(ruleAndKind.kind,
+                                            preference.ruleAndKind?.pushRule ?: ruleAndKind.pushRule,
+                                            newRule)
+                                }
+                                if (!isAdded) {
+                                    return@launch
+                                }
+                                result.onSuccess {
+                                    preference.setPushRule(ruleAndKind.copy(pushRule = newRule))
+                                }
+                                hideLoadingView()
```

Comment:
This is a detail, but now, I would rather call `hideLoadingView()` just after the check if Fragment is added.

Reply:
Technically the behaviour would be different (out of scope) but I'm not bothered if you're not bothered.

Reply:
Yes. I do not think it will introduce a bug if we reorder. Also we could use `.fold()` instead of `onSuccess` and `onFailure` which is a bit more verbose (but equivalent)